diff --git a/fs/bcachefs/fs-io-direct.c b/fs/bcachefs/fs-io-direct.c
index d5340973a1fd..43fd6ece076c 100644
--- a/fs/bcachefs/fs-io-direct.c
+++ b/fs/bcachefs/fs-io-direct.c
@@ -144,7 +144,7 @@ static int bch2_direct_IO_read(struct kiocb *req, struct iov_iter *iter)
 		bio->bi_iter.bi_sector	= offset >> 9;
 		bio->bi_private		= dio;
 
-		ret = bio_iov_iter_get_pages(bio, iter);
+		ret = bio_iov_iter_get_pages(bio, iter, 0);
 		if (ret < 0) {
 			/* XXX: fault inject this path */
 			bio->bi_status = BLK_STS_RESOURCE;
@@ -458,7 +458,7 @@ static __always_inline long bch2_dio_write_loop(struct dio_write *dio)
 		EBUG_ON(current->faults_disabled_mapping);
 		current->faults_disabled_mapping = mapping;
 
-		ret = bio_iov_iter_get_pages(bio, &dio->iter);
+		ret = bio_iov_iter_get_pages(bio, &dio->iter, 0);
 
 		dropped_locks = fdm_dropped_locks();
 
