From 51fab2f32be89abfb0282c1b79390a9b1b37ca69 Mon Sep 17 00:00:00 2001
From: Xarblu <xarblu@protonmail.com>
Date: Wed, 15 Oct 2025 09:27:05 +0200
Subject: [PATCH 8/8] block: re-export bio_iov_iter_get_pages

Needed by bcachefs
---
 block/bio.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/block/bio.c b/block/bio.c
index b3a79285c278..5d6df3e859a9 100644
--- a/block/bio.c
+++ b/block/bio.c
@@ -1360,6 +1360,7 @@ int bio_iov_iter_get_pages(struct bio *bio, struct iov_iter *iter,
 		return bio_iov_iter_align_down(bio, iter, len_align_mask);
 	return ret;
 }
+EXPORT_SYMBOL_GPL(bio_iov_iter_get_pages);
 
 static void submit_bio_wait_endio(struct bio *bio)
 {
-- 
2.51.0

